### PR TITLE
tar defaults to format=gnu, whose sparse extension breaks the tar crate

### PR DESCRIPTION
**Pull request by alsuren**
See https://github.com/alsuren/cargo-quickinstall/issues/87 for discussion.

### CODE REVIEW

File: build-version.sh [hunk 1]
```diff
@@ -57,6 +57,6 @@ done
 #
 # TODO: maybe we want to make a ~/.cargo-quickinstall/bin to untar into,
 # and add symlinks into ~/.cargo/bin, to aid debugging?
-tar -czf "${TEMPDIR}/${CRATE}-${VERSION}-${TARGET_ARCH}.tar.gz" $BINARIES
+tar --posix -czf "${TEMPDIR}/${CRATE}-${VERSION}-${TARGET_ARCH}.tar.gz" $BINARIES
```
Comment (NobodyXu) [inline build-version.sh:60]: We might have to use `--portability` here:

```suggestion
tar --portability -czf "${TEMPDIR}/${CRATE}-${VERSION}-${TARGET_ARCH}.tar.gz" $BINARIES
```

**Review by NobodyXu — CHANGES_REQUESTED** [review]

File: build-version.sh (hunk 1 above)
Comment (alsuren) [reply build-version.sh:60]: Amusingly, I couldn't see --portability in the bsdtar man page.

Comment (NobodyXu) [reply build-version.sh]: Perhaps we can use `--format=v7` instead (`--portability` seems to be same as it)?

**Review by NobodyXu — APPROVED** [review]

**Comment by alsuren** [comment]
ugh. I cheated myself out of having code coverage for this change: https://github.com/alsuren/cargo-quickinstall/runs/7915295137?check_suite_focus=true#step:4:49

**Pull request merged**